src/maple_pool/depository.ts

```diff
@@ -0,0 +1,225 @@
+
+    const depositoryCollateral = await this.findDepositoryCollateralAddress(
+      depository,
+      uxdProgramId
+    );
+
+    const mapleLender = await this.findLenderAddress(
+      maplePool,
+      depository,
+      syrupProgramId
+    );
+
+    const mapleLockedShares = await this.findLockedSharesAddress(
+      mapleLender,
+      syrupProgramId
+    );
+    const mapleLenderShares = await this.findLenderSharesAddress(
+      depository,
+      mapleSharesMint
+    );
+
+    return new MaplePoolDepository(
+      depository,
+      collateralMint,
+      collateralDecimals,
+      collateralSymbol,
+      depositoryCollateral,
+      maplePool,
+      maplePoolLocker,
+      mapleGlobals,
+      mapleLender,
+      mapleSharesMint,
+      mapleLockedShares,
+      mapleLenderShares,
+      syrupProgramId
+    );
+  }
+
+  private static async findDepositoryCollateralAddress(
+    depository: PublicKey,
+    uxdProgramId: PublicKey
+  ) {
+    return (
+      await PublicKey.findProgramAddress(
+        [
+          Buffer.from(MAPLE_POOL_DEPOSITORY_COLLATERAL_NAMESPACE),
+          depository.toBytes(),
+        ],
+        uxdProgramId
+      )
+    )[0];
+  }
+
+  private static async findLenderAddress(
+    maplePool: PublicKey,
+    depository: PublicKey,
+    syrupProgramId: PublicKey
+  ): Promise<PublicKey> {
+    return (
+      await PublicKey.findProgramAddress(
+        [
+          Buffer.from(MAPLE_INTERNAL_LENDER_NAMESPACE),
+          maplePool.toBytes(),
+          depository.toBytes(),
+        ],
+        syrupProgramId
+      )
+    )[0];
+  }
+
+  private static async findLenderSharesAddress(
+    owner: PublicKey,
+    mapleSharesMint: PublicKey
+  ): Promise<PublicKey> {
+    return (await findATAAddrSync(owner, mapleSharesMint))[0];
+  }
+
+  private static async findLockedSharesAddress(
+    mapleLender: PublicKey,
+    syrupProgramId: PublicKey
+  ): Promise<PublicKey> {
+    return (
+      await PublicKey.findProgramAddress(
+        [
+          Buffer.from(MAPLE_INTERNAL_LOCKED_SHARES_NAMESPACE),
+          mapleLender.toBytes(),
+        ],
+        syrupProgramId
+      )
+    )[0];
+  }
+
+  public info() {
+    console.groupCollapsed('[Maple Pool Depository debug info]');
+    console.table({
+      ['pda']: this.pda.toBase58(),
+      ['depositoryCollateral']: this.depositoryCollateral.toBase58(),
+      ['collateralMint']: this.collateralMint.toBase58(),
+      ['collateralDecimals']: this.collateralDecimals,
+      ['collateralSymbol']: this.collateralSymbol,
+      ['maplePool']: this.maplePool.toBase58(),
+      ['maplePoolLocker']: this.maplePoolLocker.toBase58(),
+      ['mapleGlobals']: this.mapleGlobals.toBase58(),
+      ['mapleLender']: this.mapleLender.toBase58(),
+      ['mapleSharesMint']: this.mapleSharesMint.toBase58(),
+      ['mapleLockedShares']: this.mapleLockedShares.toBase58(),
+      ['mapleLenderShares']: this.mapleLenderShares.toBase58(),
+    });
```

Comment:
hm, I just see it now kek _(have copy pasted it as well in mercurial)_

actually there are no reason for the keys to be wrapped lol

```suggestion
    console.table({
      pda: this.pda.toBase58(),
      depositoryCollateral: this.depositoryCollateral.toBase58(),
      collateralMint: this.collateralMint.toBase58(),
      collateralDecimals: this.collateralDecimals,
      collateralSymbol: this.collateralSymbol,
      maplePool: this.maplePool.toBase58(),
      maplePoolLocker: this.maplePoolLocker.toBase58(),
      mapleGlobals: this.mapleGlobals.toBase58(),
      mapleLender: this.mapleLender.toBase58(),
      mapleSharesMint: this.mapleSharesMint.toBase58(),
      mapleLockedShares: this.mapleLockedShares.toBase58(),
      mapleLenderShares: this.mapleLenderShares.toBase58(),
    });
```